Add GTSRB dataset to prototypes (#5214)
pytorch/vision: torchvision/prototype/datasets/_builtin/README.md

warm-up times when data is buffered that is only needed at runtime.

Thus, all buffered datapipes should be used as early as possible, e.g. zipping
two datapipes of file handles rather than trying to zip already loaded images.

There are two special datapipes that are not used through their class, but
through the functions `hint_sharding` and `hint_shuffling`. As the name implies
they only hint part in the datapipe graph where sharding and shuffling should
take place, but are no-ops by default. They can be imported from
`torchvision.prototype.datasets.utils._internal` and are required in each
dataset.

Finally, each item in the final datapipe should be a dictionary with `str` keys.
There is no standardization of the names (yet!).

## FAQ

### How do I start?

Get the skeleton of your dataset class ready with all 3 methods. For
`_make_datapipe()`, you can just do `return resources_dp[0]` to get started.
Then import the dataset class in
`torchvision/prototype/datasets/_builtin/__init__.py`: this will automatically
register the dataset and it will be instantiable via
`datasets.load("mydataset")`. On a separate script, try something like

> ```py

Reviewer: Oh, does this work as intended? My current understanding is that you need 

```suggestion
```python
```

Author: I think it works, if we try to edit a .md file interactively from GitHub we can see that it gets formatted properly

![image](https://user-images.githubusercontent.com/1190450/150516886-a5bae39b-a097-4f74-945b-bd2fa4991d2e.png)

![image](https://user-images.githubusercontent.com/1190450/150516918-d1f6f7cd-6451-4fbd-89f3-9fbf18b34563.png)

> d = datasets.load("mydataset")
> for e in d:
>     print(e)  # this is the content of the datapipe returned by _make_datapipe()

Reviewer: 1. Especially for an example, let's be more explicit here
2. I usually add a `break` to avoid loading the whole datapipe every time and terminating the execution manually. No strong opinion on that.

```suggestion
from torchvision.prototype import datasets

dataset = datasets.load("mydataset")
for sample in dataset:
    # this is the content of an item in datapipe returned by _make_datapipe()
    print(sample)
    break
```

# Or you can also inspect e in a debugger
```

This will give you an idea of what the first datapipe in `resources_dp`
contains. You can also do that with `resources_dp[1]` or `resources_dp[2]`
(etc.) if they exist. Then follow the instructions above to manipulate these

> datapipes and return the appropriate dict format.

Reviewer: ```suggestion
datapipes and return the appropriate dictionary format.
```
